module/zfs/vdev_raidz.c

```diff
@@ -1440,6 +1440,38 @@ vdev_raidz_child_done(zio_t *zio)
 	rc->rc_skipped = 0;
 }
 
+boolean_t
+vdev_raidz_need_resilver(vdev_t *vd, uint64_t offset, size_t psize)
```

Comment:
FYI, we did something similar for `zpool initialize`.  It's doing a similar calculation.  Once that's upstreamed, this could be implemented by calling `vdev_op_xlate`, and checking if the result's `rs_start==rs_end`.  The caller would have to loop over the child vdev's and call `vdev_op_xlate` on each of them. Not sure if that would actually be better, but thought you'd find it interesting.  It's also an example of adding a new vdev op, which I think should be done here.
https://github.com/delphix/delphix-os/commit/f08e5bc9008352c28922febd9f88626a1544b4dd#diff-826708a55165ad963711ccae15edaad6R2407